src/lib.rs

```diff
@@ -589,21 +672,38 @@ impl Consul {
             .body(body);
         let req = req.map_err(ConsulError::RequestError)?;
         let mut span = crate::hyper_wrapper::span_for_request(&self.tracer, &req);
+
+        let method = req.method().clone();
+        record_request_metric_if_enabled(&method, request_name);
         let future = self.https_client.request(req);
 
+        let step_start_instant = Instant::now();
         let response = if let Some(dur) = duration {
             match timeout(dur, future).await {
-                Ok(resp) => resp.map_err(ConsulError::ResponseError)?,
-                Err(_) => return Err(ConsulError::TimeoutExceeded(dur)),
+                Ok(resp) => resp.map_err(ConsulError::ResponseError),
+                Err(_) => Err(ConsulError::TimeoutExceeded(dur)),
```

Comment:
```suggestion
                Err(_) => Err(ConsulError::TimeoutExceeded(dur)),
```
why no return now?

Reply:
oh I guess metrics?

Reply:
Yeah, returning here would prevent failure and duration metrics from being recorded if there's a TimeoutExceeded error